lib/src/revset.rs

```diff
@@ -1546,33 +1546,25 @@ where
     }
 }
 
-pub struct RevsetIterator<'revset, 'index: 'revset> {
-    inner: Box<dyn Iterator<Item = IndexEntry<'index>> + 'revset>,
+pub trait RevsetIteratorExt<'index, I> {
+    fn commit_ids(self) -> RevsetCommitIdIterator<I>;
+    fn commits(self, store: &Arc<Store>) -> RevsetCommitIterator<I>;
```

Comment:
(no action needed)

These two can be implemented by `.map(...)`, but I have no idea how to express `-> impl Iterator<..>` in trait.

Reply:
Neither do I. Maybe we could return a `Box<dyn Iterator<...>>`. Anyway, I don't think this PR makes it worse, so I'll just leave that for later (if at all).